tests/version.test.js

```diff
@@ -0,0 +1,104 @@
+jest.mock("@actions/github");
+
+const { GitHub, context } = require("@actions/github");
+const setInputs = require("./test-utils");
+
+const getNextVersion = require("../src/version");
+
+const buildCommitResponse = (messages) => {
+  const commits = messages.map((message) => {
+    return { commit: { message } };
+  });
+  return {
+    data: commits,
+  };
+};
+
+const mockGitHub = (messages) => {
+  const commits = buildCommitResponse(messages);
+  const listCommits = jest.fn().mockReturnValueOnce(commits);
+  const listRefs = jest
+    .fn()
+    .mockReturnValueOnce(
+      Promise.resolve({
+        data: [
+          { ref: "refs/tags/1.2.0" },
+          { ref: "refs/tags/1.1.1" },
+          { ref: "refs/tags/1.1.0" },
+          { ref: "refs/tags/1.2.1-rc.0" },
```

Comment:
You could add another test set with data that doesn't align with the setup to check the behaviour if this is used in an old repository with already existing version tags. For example "1.2.0a", "1.2.1-alpha", "1.2.1-rc".

Reply:
The existing prerelease format doesn't really matter because this uses the first version bump that is available so all of your examples would be just ignored. I think the two RC version tests cover this use case. The only requirement is that the existing version format should also follow the major.minor.patch scheme, otherwise versioning will start from 0.0.1. Not sure if this is worth adding a test though.

Reply:
Ack. Maybe you could add that as a note for the documentation.